tests/integration/devfile/cmd_devfile_push_test.go

```diff
@@ -188,14 +188,23 @@ var _ = Describe("odo devfile push command tests", func() {
 					}, 180, 10).Should(Equal(true))
 					newPod := commonVar.CliRunner.GetRunningPodNameByComponent(cmpName, commonVar.Project)
 					commonVar.CliRunner.PodsShouldBeRunning(commonVar.Project, newPod)
-
 				})
 
 				It("should execute run command on odo push", func() {
 					output = helper.Cmd("odo", "push", "--project", commonVar.Project).ShouldPass().Out()
 					Expect(output).To(ContainSubstring("Executing devrun command"))
 				})
 			})
+
+			When("the Deployment's Replica count (pods) is scaled to 0", func() {
+				BeforeEach(func() {
+					commonVar.CliRunner.ScalePodToZero(cmpName, "app")
```

Comment:
I think you will need to wait for the pod to be terminated before to continue, or the scale command won't have the time to have any effect